### PR TITLE
feat(add comment) - TS

### PR DESCRIPTION
**Pull request by shakederin**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  "author": "",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/shakederin/Instagram-post-clone/issues"
+  },
+  "homepage": "https://github.com/shakederin/Instagram-post-clone#readme",
+  "devDependencies": {
+    "typescript": "^4.6.4"
+  }
+}
```
Comment (YardenPorat) [inline package.json:23]: eof

File: public/index.html [hunk 2]
```diff
@@ -92,6 +92,7 @@
             </div>
         </div>
     </div>
+    <script src="../dist/index.js" type="module"></script>
 </body>
 
 </html>
```
Comment (YardenPorat) [inline public/index.html:98]: eof

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { makeComment } from "./utils/createElements.js";
+const inputElement = document.getElementById("commentInput");
```
Comment (YardenPorat) [inline src/index.ts:2]: new line

File: src/index.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+) {
+  formElement.addEventListener("submit", (event) => {
+    event.preventDefault();
+    const newMsg = inputElement.value;
+    if (newMsg && ulElement) {
+      ulElement.append(makeComment(ppURL, "tonytoscani", newMsg));
+      inputElement.value = "";
+    }
+  });
+}
```
Comment (YardenPorat) [inline src/index.ts:20]: insert to a function

File: src/index.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { makeComment } from "./utils/createElements.js";
+const inputElement = document.getElementById("commentInput");
+const formElement = document.getElementById("inputForm");
+const ulElement = document.getElementById("commentsList");
+const ppURL =
```
Comment (YardenPorat) [inline src/index.ts:5]: pp unclear

**Comment by Ari-Kishon** [comment]
We talked about this FOUR times!

Dont open a new pr , use the old one....

File: src/index.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+if (
+  formElement instanceof HTMLFormElement &&
+  inputElement instanceof HTMLInputElement
+) {
+  formElement.addEventListener("submit", (event) => {
+    event.preventDefault();
+    const newMsg = inputElement.value;
+    if (newMsg && ulElement) {
+      ulElement.append(makeComment(ppURL, "tonytoscani", newMsg));
```
Comment (YardenPorat) [inline src/index.ts:16]: make -> create
keep same conventions throughout the code

File: src/index.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  "https://cdn.pixabay.com/photo/2021/05/20/11/58/beauty-6268460_1280.jpg";
+
+if (
+  formElement instanceof HTMLFormElement &&
+  inputElement instanceof HTMLInputElement
+) {
+  formElement.addEventListener("submit", (event) => {
+    event.preventDefault();
+    const newMsg = inputElement.value;
+    if (newMsg && ulElement) {
```
Comment (YardenPorat) [inline src/index.ts:15]: Is it a comment or a message?

File: src/utils/helpers.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
+export const createNewElement = (
```
Comment (YardenPorat) [inline src/utils/helpers.ts:1]: helpers\utils file - antipattern

File: package.json [hunk 9]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "instagram-post-clone",
+  "description": "",
```
Comment (YardenPorat) [inline package.json:3]: delete, or fill

File: package.json [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+  "scripts": {
+    "build": "tsc",
+    "watch": "tsc -w"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/shakederin/Instagram-post-clone.git"
+  },
+  "keywords": [],
+  "author": "",
```
Comment (YardenPorat) [inline package.json:14]: delete, or fill

Comment (YardenPorat) [reply package.json:14]: same for other keys in the file

File: src/utils/createElements.ts [hunk 11]
```diff
@@ -0,0 +1,60 @@
+import { createNewElement, likeCommentClicked } from "./helpers.js";
+
+const createCommentText = (userName: string, comment: string) => {
+  const commentHeaderElement = createNewElement("div", "commentHeader", "");
```
Comment (YardenPorat) [inline src/utils/createElements.ts:4]: why not h1-h6?

File: src/utils/createElements.ts [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+  const commentHeaderAndContentElement = document.createElement("div");
```
Comment (YardenPorat) [inline src/utils/createElements.ts:5]: why not header?
why not header element?

File: src/utils/createElements.ts [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+  const commentBodyElement = createNewElement("span", "", comment);
```
Comment (YardenPorat) [inline src/utils/createElements.ts:6]: remove all argument that are passed as an empty string, use an object

File: src/utils/createElements.ts [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+  const commentHeaderBoldElement = createNewElement("b", "", `${userName} `);
```
Comment (YardenPorat) [inline src/utils/createElements.ts:7]:
```
`${userName} `)
```
don't style with text

Comment (YardenPorat) [reply src/utils/createElements.ts:7]: `commentHeaderBoldElement` ->  `commenterName` or similar

File: src/utils/createElements.ts [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+  commentHeaderAndContentElement.append(
+    commentHeaderBoldElement,
+    commentBodyElement
+  );
+  commentHeaderElement.append(commentHeaderAndContentElement);
```
Comment (YardenPorat) [inline src/utils/createElements.ts:12]: same

File: src/utils/createElements.ts [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+  return commentHeaderElement;
+};
+
+const createCommentExraInfo = () => {
+  const commentInfoElement = createNewElement("div", "commentInfo", "");
+  const hourAgoElement = createNewElement("div", "infoSubtitle", "10 h");
+  const likesInfoElement = createNewElement("b", "infoSubtitle", "like");
+  const replayInfoElement = createNewElement("b", "infoSubtitle", "Reply");
```
Comment (YardenPorat) [inline src/utils/createElements.ts:20]: when you use `b` you use it for semantic markup and not styling

Comment (YardenPorat) [reply src/utils/createElements.ts:20]: likesCount
replayInfoElement -> likeButton

File: src/utils/createElements.ts [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+  }
+  return commentSvgElement;
+};
+
+export const makeComment = (
+  profilePicture: string,
+  userName: string,
+  comment: string
+) => {
+  const LIElement = createNewElement("li", "comment", "");
```
Comment (YardenPorat) [inline src/utils/createElements.ts:46]: liElement

File: tsconfig.json [hunk 18]
```diff
@@ -0,0 +1,101 @@
...
+    // "emitDecoratorMetadata": true,                    /* Emit design-type metadata for decorated declarations in source files. */
+    // "jsxFactory": "",                                 /* Specify the JSX factory function used when targeting React JSX emit, e.g. 'React.createElement' or 'h' */
+    // "jsxFragmentFactory": "",                         /* Specify the JSX Fragment reference used for fragments when targeting React JSX emit e.g. 'React.Fragment' or 'Fragment'. */
+    // "jsxImportSource": "",                            /* Specify module specifier used to import the JSX factory functions when using `jsx: react-jsx*`.` */
+    // "reactNamespace": "",                             /* Specify the object invoked for `createElement`. This only applies when targeting `react` JSX emit. */
+    // "noLib": true,                                    /* Disable including any library files, including the default lib.d.ts. */
+    // "useDefineForClassFields": true,                  /* Emit ECMAScript-standard-compliant class fields. */
+
+    /* Modules */
+    "module": "ES6",                                /* Specify what module code is generated. */
```
Comment (YardenPorat) [inline tsconfig.json:27]: I think you can use something newer

File: tsconfig.json [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+    // "noImplicitOverride": true,                       /* Ensure overriding members in derived classes are marked with an override modifier. */
+    // "noPropertyAccessFromIndexSignature": true,       /* Enforces using indexed accessors for keys declared using an indexed type */
+    // "allowUnusedLabels": true,                        /* Disable error reporting for unused labels. */
+    // "allowUnreachableCode": true,                     /* Disable error reporting for unreachable code. */
+
+    /* Completeness */
+    // "skipDefaultLibCheck": true,                      /* Skip type checking .d.ts files that are included with TypeScript. */
+    "skipLibCheck": true                                 /* Skip type checking all .d.ts files. */
+  }
+}
```
Comment (YardenPorat) [inline tsconfig.json:101]: eof

**Pull request closed without merging**